### PR TITLE
Add feature handle group

### PR DESCRIPTION
**Pull request by enrinal**
#17 
In this feature can send massage based on gitlab group project, using helper GroupDetection() **see helper**

Work well with push request and merge request

### CODE REVIEW

**Comment by enrinal** [comment]
Work well in push and merge request

- In push request, helper read Repository URL and split url string, in pushhandler() there is a switch case that receives a return from the helper and sends it according to the workplace group
- In merge request, helper read SSH URL and split url string, in mergehandler()  there is a switch case that receives a return from the helper and sends it according to the workplace group

*Please change Thread channel in each helper before using it in production*

Q: Why push and merge has different input variable ? 
A: Because push and merge has different struct, so possible way to get group information by repository url in push request, and ssh url in merge request

File: workplace-incoming-hook.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ var (
 	ThreadGitlab        string
 	ThreadTuleap        string
 	ThreadAppCenter     string
+	ThreadBackend       string
```
Comment (jojoarianto) [inline workplace-incoming-hook.go]: it's not allowed here bro. Please use dynamic mapping

File: workplace-incoming-hook.go [hunk 2]
```diff
@@ -360,7 +362,17 @@ func PushHandler(body string) {
 			// Third line (last commit message)
 			message += "```" + MessageEncode(lastCommit.Message) + "```"
 		}
-		SendWorkchatMessage(ThreadGitlab, message, ChatType)
+		// SendWorkchatMessage(ThreadGitlab, message, ChatType)
+		switch gr := helper.GroupDetection(j.Repository.URL); gr {
+		case "back-end":
+			SendWorkchatMessage(ThreadBackend, message, ChatType) // Change ThreadTuleap to your group Thread
```
Comment (jojoarianto) [inline workplace-incoming-hook.go]: it's not allowed here bro, this repository must have general purpose only. or you can create new configuration for mapping thread in new configuration file.

**Review by jojoarianto — CHANGES_REQUESTED** [review]

File: workplace-incoming-hook.go (hunk 1 above)
Comment (enrinal) [reply workplace-incoming-hook.go]: Ok , I will change it

File: workplace-incoming-hook.go (hunk 2 above)
Comment (enrinal) [reply workplace-incoming-hook.go]: Ok , I will change it

File: helper/groupdetection.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+		This format using this url format "git@gitlab.warungpintar.co:back-end/gitlab-test.git"
+
+		Flow :
+		1. Split git@gitlab.warungpintar.co:back-end/gitlab-test.git to [ back-end/gitlab-test.git]
+		2. Split [ back-end/gitlab-test.git] to [back-end gitlab-test.git]
+		3. Get string in slice[0] which is backend
+
+	*/
+	link := strings.Split(url, "git@gitlab.warungpintar.co:") //change git@gitlab.warungpintar.co: tou your own url
```
Comment (jojoarianto) [inline helper/groupdetection.go]: it's not allowed here bro

Comment (jojoarianto) [reply helper/groupdetection.go]: please change with variable. and make it more dynamic for another company too.

*nb:you can put it in config.json

**Review by jojoarianto — CHANGES_REQUESTED** [review]

File: helper/groupdetection.go (hunk 3 above)
Comment (enrinal) [reply helper/groupdetection.go]: Bro, I forgot to remove that helper, GroupDetection helper is unnecessary, already remove that helper in new Push

**Review by jojoarianto — APPROVED** [review]
Look's Good To Me

**Pull request merged**